aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -2234,6 +2234,33 @@ Tensor nonzero_cpu(const Tensor& self) {
   return result;
 }
 
+Tensor nonzero_static(
+    const Tensor& self,
+    int64_t size,
+    const Scalar& fill_value) {
+  // Check if self is not 0-d tensor
+  TORCH_CHECK(self.dim() > 0, "Rank must be positive");
+  // Check if self is not 0-d tensor
+  TORCH_CHECK(size >= 0, "'size' must be non-negative");
+  // Check if fill_value type is integer types
+  TORCH_CHECK(
+      fill_value.isIntegral() == true, "'fill_value' must be integer types");
+  if (size == 0) {
+    // Return an empty tensor
+    return at::empty(0, ScalarType::Long);
+  }
+  // Delegate call to regular nonzero to get a data-dependent output
+  auto dyn_result = at::nonzero(self);
```

Comment:
No. If you're doing this in C++, you must implement this properly (with no syncs). The whole reason I told you to register it via Python dispatcher was so that the crappy implementation wasn't available by default to users. People will hit this version, and will reasonable start asking questions about why it is syncing even with an explicit size.